internal/network/command_network_link_endpoint_list.go

```diff
@@ -0,0 +1,99 @@
+package network
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	pcmd "github.com/confluentinc/cli/v3/pkg/cmd"
+	"github.com/confluentinc/cli/v3/pkg/errors"
+	"github.com/confluentinc/cli/v3/pkg/output"
+)
+
+type listNetworkLinkEndpointOut struct {
+	Id                 string `human:"ID" serialized:"id"`
+	Name               string `human:"Name" serialized:"name"`
+	Network            string `human:"Network" serialized:"network"`
+	Description        string `human:"Description,omitempty" serialized:"description,omitempty"`
+	NetworkLinkService string `human:"Network Link Service" serialized:"network_link_service"`
+	Phase              string `human:"Phase" serialized:"phase"`
+}
+
+func (c *command) newNetworkLinkEndpointListCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List network link endpoints.",
+		Args:  cobra.NoArgs,
+		RunE:  c.networkLinkEndpointList,
+	}
+
+	cmd.Flags().String("name", "", "Network Link endpoint display name.")
+	addNetworkFlag(cmd, c.AuthenticatedCLICommand)
+	c.addNetworkLinkServiceFlag(cmd)
+	addNetworkLinkEndpointPhaseFlag(cmd)
+	pcmd.AddContextFlag(cmd, c.CLICommand)
+	pcmd.AddEnvironmentFlag(cmd, c.AuthenticatedCLICommand)
+	pcmd.AddOutputFlag(cmd)
+
+	return cmd
+}
+
+func (c *command) networkLinkEndpointList(cmd *cobra.Command, _ []string) error {
+	endpoints, err := c.getNetworkLinkEndpoints()
+	if err != nil {
+		return err
+	}
+
+	name, err := cmd.Flags().GetString("name")
+	if err != nil {
+		return err
+	}
+
+	network, err := cmd.Flags().GetString("network")
+	if err != nil {
+		return err
+	}
+
+	service, err := cmd.Flags().GetString("network-link-service")
+	if err != nil {
+		return err
+	}
+
+	phase, err := cmd.Flags().GetString("phase")
+	if err != nil {
+		return err
+	}
+
+	list := output.NewList(cmd)
+	for _, endpoint := range endpoints {
+		if endpoint.Spec == nil {
+			return fmt.Errorf(errors.CorruptedNetworkResponseErrorMsg, "spec")
+		}
+		if endpoint.Status == nil {
+			return fmt.Errorf(errors.CorruptedNetworkResponseErrorMsg, "status")
+		}
+
+		if name != "" && endpoint.Spec.GetDisplayName() != name {
+			continue
+		}
+		if network != "" && endpoint.Spec.Network.GetId() != network {
+			continue
+		}
+		if service != "" && endpoint.Spec.NetworkLinkService.GetId() != service {
+			continue
+		}
+		if phase != "" && endpoint.Status.GetPhase() != phase {
+			continue
+		}
```

Comment:
I assume you are trying to filter the endpoints based on the flags?

The list API actually supports filtering with query parameters. There was an issue for `MultipleSearchFilter` in the SDK so I didn't add filtering for our LIST commands.

Reply:
https://docs.confluent.io/cloud/current/api.html#tag/Network-Link-Endpoints-(networkingv1)/operation/listNetworkingV1NetworkLinkEndpoints

1. You would need to update the SDK version to ([networking/v0.9.0](https://github.com/confluentinc/ccloud-sdk-go-v2/releases/tag/networking%2Fv0.9.0) and [networking-ip/v0.2.0](https://github.com/confluentinc/ccloud-sdk-go-v2/releases/tag/networking-ip%2Fv0.2.0))
2. Update the flags based on the API Spec (some of them should be `[]string`)
3. Update the `getNetworkLinkEndpoints` function (see [ref](https://github.com/confluentinc/ccloud-sdk-go-v2/blob/master/networking/v1/docs/NetworkLinkEndpointsNetworkingV1Api.md#listnetworkingv1networklinkendpoints ) to learn how to use the filters in the SDK)

Reply:
In [networking/v0.9.0](https://github.com/confluentinc/ccloud-sdk-go-v2/releases/tag/networking%2Fv0.9.0), the workaround for `MultipleSearchFilter` resulted in a few models to use `Set` type. Of the updated models,`ZonesInfo` is not compatible with type `Set`. I've removed flags here, will add them [separately](https://confluentinc.atlassian.net/browse/TRAFFIC-9348) once SDK is [updated](https://confluent.slack.com/archives/C04PBPC4H1T/p1701712403845029). CC @syoung-cflt @ejagade